docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,356 @@
+```
+
+### oras push
+
+When using `oras pull` with the flag `--format`, the following fields should be formatted into JSON output:
+
+- `tags`: pushed tags, this property can be one or multiple tags
+- `reference`: full artifact reference by digest, e.g, `$REGISTRY/$REPO@$DIGEST`
+- `mediaType`: media type of the pushed file (layer) 
+- `digest`: digest of the pushed file (layer) 
+- `size`: file size in bytes
+- `annotations`: contains arbitrary metadata for the image manifest
+
+For example, push two files to a repository and show the descriptor of the image manifest in pretty JSON format. 
+
+```bash
+oras push $REGISTRY/$REPO:$TAG sbom.spdx vul-scan.json --format json 
+```
+
+```json
+{
+  "tags": "$REGISTRY/$REPO:$TAG",
+  "reference": "localhost:5000/oras@sha256:4a5b8c83d153f52afdfcb422db56c2349aae3bd5ecf8338a58353b5eb6681c45",
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "digest": "sha256:4a5b8c83d153f52afdfcb422db56c2349aae3bd5ecf8338a58353b5eb6681c45",
+  "size": 820,
+  "annotations": {
+    "org.opencontainers.image.created": "2023-12-15T09:41:54Z"
+  }
+}
+```
+
+> [!NOTE]
+> When pushing a folder to filesystem, the output fields are the same as the example of pushing files as above.
+
+Push a folder to a repository and filter out the value of `reference` and `artifactType` of the pushed artifact in the standard output.
+
+```bash
+oras push $REGISTRY/$REPO:$TAG sample-folder --format '{{.reference}}, {{.mediaType}}'
+```
+
+```console
+localhost:5000@sha256:85438e6598bf35057962fff34399a362d469ca30a317939427fca6b7a289e70d, application/vnd.oci.image.manifest.v1+json
+```
+
+### oras attach
+
+When using `oras pull` with the flag `--format`, the following fields should be formatted into JSON output:
+
+- `tags`: attached tags as reference, it can be one or multiple tags
+- `reference`: full reference by digest of the referrer file
+- `mediaType`: media type of the referrer
+- `size`: referrer file size in bytes
+- `digest`: digest of the attached referrer file
+- `artifactType`: artifact type of the referrer
+- `annotations`: contains arbitrary metadata in a referrer
+
+For example, attach two files to an image and show the descriptor metadata of the referrer in JSON format.
+
+```bash
+oras attach $REGISTRY/$REPO:$TAG --artifact-type example/report-and-sbom vul-report.json:example/vul-scan sbom.spdx:example/sbom --format json
+```
+
+```json
+{
+  "tags": "$REGISTRY/$REPO:$TAG",
```

Comment:
Is this comma separated or an array?

Reply:
It a separated comma at the end of each line in formatted JSON output. For this line, it can be one or multiple tags either.

Reply:
IMHO making this attribute an array will make auotmation easier.

Reply:
@qweeah That's a good point. I think make `tags` be an array would be much convenient and flexible for users. They can choose to filter one tag or all tags from array.